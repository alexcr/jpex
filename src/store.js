import React from 'react';
import { useLocalStore } from 'mobx-react-lite';
import { createStore } from './createStore';

const storeContext = React.createContext(null);

export const StoreProvider = ({children}) => {
  const store = useLocalStore(createStore);
  return (
    <storeContext.Provider value={store}>
      {children}
    </storeContext.Provider>
  );
}

export const useStore = () => {
  const store = React.useContext(storeContext);
  if (!store) {
    throw new Error('useStore must be used within a StoreProvider.');
  }
  return store;
}
