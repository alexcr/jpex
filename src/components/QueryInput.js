import React from 'react';
import styled from 'styled-components';
import { observer } from 'mobx-react-lite';
import { DelayInput } from 'react-delay-input';
import { useStore } from '../store';

const StyledInput = styled(DelayInput)`
  box-sizing: border-box;
  width: 100%;
  padding: 15px;
  border: 0px;
  height: 50px;
  font-size: 20px;
  border-top-left-radius: 15px;
  border-top-right-radius: 15px;
  border: 1px solid #006616;
  font-family: 'Montserrat', sans-serif;

  &:focus {
    outline: none;
  }

  &.invalid-true {
    color: red;
  }
`;

export const QueryInput = observer(() => {
  const store = useStore();
  return (
    <StyledInput
      delayTimeout={200}
      type="text"
      className={`query invalid-${store.invalidQuery}`}
      value={store.query}
      onChange={e => store.setQuery(e.target.value)}
      placeholder="Query JSONPath..."
      disabled={store.invalidJson}
    />
  );
});
