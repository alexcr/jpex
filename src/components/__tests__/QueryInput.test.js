import React from 'react';
import { render } from 'enzyme';
import { QueryInput } from '../QueryInput';
import { StoreProvider } from '../../store';

describe('QueryInput', () => {
  it('should render the query input correctly', () => {
    expect(
      render(<StoreProvider><QueryInput /></StoreProvider>)
    ).toMatchSnapshot();
  });
});
