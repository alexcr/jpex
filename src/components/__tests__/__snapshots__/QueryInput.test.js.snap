// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryInput should render the query input correctly 1`] = `
<input
  class="sc-bdnylx jFCUOZ query invalid-false"
  placeholder="Query JSONPath..."
  type="text"
  value=""
/>
`;
